flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java

```diff
@@ -100,6 +111,27 @@ public void init() throws Exception {
 		getEnvironment().getMetricGroup().gauge(MetricNames.IO_CURRENT_INPUT_WATERMARK, this.inputWatermarkGauge::getValue);
 	}
 
+	private StreamTaskInput<IN> wrapWithSorted(StreamTaskInput<IN> input) {
```

Comment:
What about other tasks? I guess this PR is targeting just the one input? If so, maybe add `throw new UnsupportedOperationException()` in other tasks, to have a bit more "releasable master branch" while we are awaiting the follow ups?

Reply:
Yes, the other tasks will be handled in separate PRs (I already opened a draft for two input: https://github.com/apache/flink/pull/13529 which lacks mostly comments).

Sure I will add exceptions, even though I think it is in a releasable shape already. The feature is not exposed to users, there is no straightforward way to enable it. The only way is to edit the StreamGraph manually, as I am doing in a test, but that's a hack.